source/gateway.go

```diff
@@ -220,19 +217,41 @@ func (sc *gatewaySource) setResourceLabel(config istiomodel.Config, endpoints []
 }
 
 func (sc *gatewaySource) targetsFromIstioIngressStatus() (targets endpoint.Targets, err error) {
-	if svc, e := sc.kubeClient.CoreV1().Services(sc.istioNamespace).Get(sc.istioIngressGatewayName, metav1.GetOptions{}); e != nil {
-		err = e
-	} else {
-		for _, lb := range svc.Status.LoadBalancer.Ingress {
-			if lb.IP != "" {
-				targets = append(targets, lb.IP)
+	if len(sc.istioIngressGateways) == 0 || (len(sc.istioIngressGateways) == 1 && sc.istioIngressGateways[0] == "") {
+		if svcs, e := sc.kubeClient.CoreV1().Services("").List(metav1.ListOptions{}); e != nil {
+			err = e
+		} else {
+			for _, svc := range svcs.Items {
+				for _, lb := range svc.Status.LoadBalancer.Ingress {
```

Comment:
I don’t know exactly but this looks like a nil ptr if it’s not a service type loadbalancer.